engineering_tools/carma-config

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+
+#  Copyright (C) 2018-2019 LEIDOS.
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#  use this file except in compliance with the License. You may obtain a copy of
+#  the License at
+# 
+#  http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations under
+#  the License.
+
+# Code below largely based on template from Stack Overflow:
+# https://stackoverflow.com/questions/37257551/defining-subcommands-that-take-arguments-in-bash
+# Question asked by user 
+# DiogoSaraiva (https://stackoverflow.com/users/4465820/diogosaraiva)
+# and answered by user 
+# Charles Duffy (https://stackoverflow.com/users/14122/charles-duffy)
+# Attribution here is in line with Stack Overflow's Attribution policy cc-by-sa found here:
+# https://stackoverflow.blog/2009/06/25/attribution-required/
+
+carma-config__set() {
+    if [[ -z $1 ]]; then
+        echo "Please specify a tag string for carma-config to set."
+        echo "Done."
+        exit -1
+    fi
+
+    if docker container inspect carma-config > /dev/null; then
+        echo "Clearing existing CARMA configuration instance..."
+        docker rm carma-config
+    fi
+
+    docker run --name carma-config $1
+    echo "Set $1 as current CARMA configuration instance..."
+}
+
+carma-config__edit() {
+    if ! docker container inspect carma-config > /dev/null 2>&1; then
+        echo "No existing CARMA configuration found, nothing to edit. Please set a config."
+        echo "Done."
+        exit 1
+    fi
+
+    echo "Opening shell inside carma-config container with read/write privileges..."
+    docker run -it --rm --volumes-from carma-config usdotfhwastol/carma-base:latest
```

Comment:
Will the carma-base version of "latest" always generate the correct version or do we need to version this call?

Reply:
This is just a temporary container (the `--rm` arg even ensures that the container is deleted after the process exits) so we can get a bash shell and our dev tools in an environment where the carma-config container is mounted. It should be fine here as the only thing that'll likely be used is vim/nano/etc so the version of the image doesn't really mater much